### ISSUE TITLE
spring security false positives for version 1.3.0

### ISSUE THREAD

**Issue by sellersj** [dependency-check/DependencyCheck#311]
org.springframework.security:spring-security-core versions 3.2.6.RELEASE and 4.0.2.RELEASE are being incorrectly flagged for:
CVE-2014-3578
CVE-2014-3625

The fix from https://github.com/jeremylong/DependencyCheck/issues/162 no longer seems to be working.

**Comment by sellersj** [comment]
I'm not sure, but maybe this is because the company name changed from vmware to pivotal? At least as far as the security issues go.

**Comment by sellersj** [comment]
I've added an entry to the suppression file, but it doesn't seem to effect the output.

When using the maven plugin, it does not show this library as an issue. If I use the CLI it does.

**Comment by jeremylong** [comment]
Thanks for the bug report - any chance you could provide the HTML (or XML) report from the CLI and Maven plugin with the differences in findings?

Regarding the vmware vs. springsource vs. pivotal - I have no idea. I know VMware bought springsource a while back. I have no idea how pivotal fits in as I don't follow Spring. I do know that CVEs use all three (sometimes for the same version).

**Comment by jeremylong** [comment]
The false positive has been corrected in the core. However, I really would like more information on the spring-security being flagged as vulnerable by the CLI but not via the maven plugin. Is there any chance you could provide more details about this? How is the dependency declared in the pom.xml and how was the dependency-check-maven plugin to the project?

**Comment by sellersj** [comment]
Thanks for the fix!

Sorry for the delay. I was away from work.
- Both the maven task and cli connect to nexus OSS 2.11.4-01
- The dependency on spring-security is typically done in a corporate BOM, but I've reproduced the issue without using the BOM.
- The dependency-check-maven plugin to the project is configured in the dependencyManagement section and "active" via a maven profile
- This is reproducible on linux and Windows 7
- The project is of type "war" so that any dependent libs are under target/ and found via the cli tool.

Here is the evidence for the CLI

```
Source  Name    Value
file    name    4.0.2
file    name    spring-security-core-4.0.2.RELEASE
hint analyzer   product springsource_spring_framework
hint analyzer   vendor  pivotal
hint analyzer   vendor  vmware
jar package name    core
jar package name    security
jar package name    springframework
Manifest    Implementation-Title    spring-security-core
Manifest    Implementation-Version  4.0.2.RELEASE
nexus   artifactid  spring-security-core
nexus   groupid org.springframework.security
nexus   version 4.0.2.RELEASE
pom artifactid  spring-security-core
pom description spring-security-core
pom groupid springframework.security
pom name    spring-security-core
pom organization name   http://spring.io/
pom version 4.0.2.RELEASE
```

Here is the evidence for the maven plugin

```
Source  Name    Value
file    name    spring-security-core-4.0.2.RELEASE
jar package name    core
jar package name    security
jar package name    springframework
Manifest    Implementation-Title    spring-security-core
nexus   artifactid  spring-security-core
nexus   groupid org.springframework.security
pom artifactid  spring-security-core
pom description spring-security-core
pom groupid org.springframework.security
pom groupid springframework.security
pom name    spring-security-core
pom organization name   http://spring.io/ 
```

The CLI report has 2 additional identifiers too:
cpe: cpe:/a:pivotal:spring_framework:4.0.2   Confidence:HIGHEST  
cpe: cpe:/a:vmware:springsource_spring_security:4.0.2   Confidence:LOW 

I can send you the generated files if you'd still like them. Would email work?
Or if it would be better, I can strip out of corporate structure and just make a git repo that should demo this issue.

**Comment by jeremylong** [comment]
After looking at the evidence you provided I have one question - what version of dependency-check-maven is being used? That looks like pre-1.3.0 evidence (as all evidence collected is output in 1.3.0 whereas previous versions only output the evidence used in finding a match).

Regardless, this issue has been resolved in 1.3.1-SNAPSHOT (tested this morning).

**Comment by sellersj** [comment]
Ah, you're right. In my test app it was using 1.2.11 for the maven plugin. I updated it to 1.3.0 and it correctly did not flag spring-security as a problem - same result as before.

I downloaded, built, and tested the code using 1.3.1-SNAPSHOT and the issue has been resolved.

Thanks!
